Core/Src/Custom/proxSensHandler.c

```diff
@@ -55,6 +69,18 @@ uint8_t vProxSensMeasure(void){
 
 }
 
+/*Function checking if there is an obstacle in set distance from sensor based on formula provided in HC-SR04 sensor datasheet*/
+bool bIsObstacleInRange(uint8_t iMsMeasureTime){
+	uint8_t iMeasuredDistance;
 
+	iMeasuredDistance = (iMsMeasureTime*1000)/58;  //Time is multiplicated by 1000 since formula requires nanoseconds, 58 is directly from the forumula 1ms ~ 17cm
 
+	if(iMeasuredDistance < TRIGGER_DISTANCE){
 
+		return true;
+	}
+	else{
+		//vLEDOff();
+		return false;
+	}
```

Comment:
iMeasuredDistance can be defined in one line `uint8_t iMeasuredDistance = (iMsMeasureTime*1000)/58;`
instead of the if statement, simply return the value `return iMeasuredDistance < TRIGGER_DISTANCE`